File: experiments/clj-vulkan/llvm/vulkan.c
#include <GLFW/glfw3.h>
#include <vulkan/vulkan.h>

long fib(int i) {
  if (i == 0) {
      return 0;
    } else if (i == 1) {
      return 1;
    } else {
    return fib(i - 1) + fib(i - 2);
  }
}
    
int main() {
  return fib(9);
}

File: experiments/host-evaluation/julia/src/vulkan.jl
#module Vulkan

function toglfwname(name)
    bits = split(name, r"-")
    reduce(*, map(uppercasefirst, bits), init="glfw")
end

function init()
    ccall(("glfwInit", "libglfw.so"), Cvoid, ())
end

t = Ptr{Cvoid}

function createwindow(width, height, name)
    @eval ccall(("glfwCreateWindow", "libglfw.so"), $t, (Int32, Int32, Cstring, Ptr{Cvoid}, Ptr{Cvoid},), $width, $height, $name, C_NULL, C_NULL)
end
